### ISSUE TITLE
gamescope-plus 3.14.29-plus1 not able to display overlay in steam-plus

### ISSUE THREAD

**Issue by philmmanjaro** [ShadowBlip/OpenGamepadUI#397]
Seems gamescope-plus 3.14.29-plus1 doesn't show OGUI overlay in steam-plus session. When I downgrade to 3.14.24-plus1 it works

**Comment by pastaq** [comment]
I was unable to reproduce after manually updating gamescope-plus and OpenGamepadUI to the latest versions.

**Comment by philmmanjaro** [comment]
Must be some sort of glitch. Now works on my end also. Have to check what issues that can be. When it doesn't work I get the left Menu open as if I would have pressed GUIDE.

**Comment by pastaq** [comment]
After watching the video it looks like a bug that I thought was solved in the state refactoring. It seems that there is a time delay from when the button is pressed to when the default profile is switched. If you press the button multiple times (the number of times doesn't matter, it's time based) it will open the menu after about 1s. This is consistent with what I've seen in the past. It will likely manifest after/while you've running a game and not on the main menu before that.

**Comment by philmmanjaro** [comment]
Yes. Some like that. Gamescope-Plus version is unrelated to that.

**Comment by philmmanjaro** [comment]
On a bad state I get happy triggers in evtest:

```
Event: time 1725338725.656849, -------------- SYN_REPORT ------------
Event: time 1725338728.468155, type 1 (EV_KEY), code 708 (BTN_TRIGGER_HAPPY5), value 1
Event: time 1725338728.468155, -------------- SYN_REPORT ------------
Event: time 1725338728.654430, type 1 (EV_KEY), code 708 (BTN_TRIGGER_HAPPY5), value 0
Event: time 1725338728.654430, -------------- SYN_REPORT ------------
Event: time 1725338764.021574, type 1 (EV_KEY), code 710 (BTN_TRIGGER_HAPPY7), value 1
Event: time 1725338764.021574, -------------- SYN_REPORT ------------
Event: time 1725338764.207614, type 1 (EV_KEY), code 710 (BTN_TRIGGER_HAPPY7), value 0
```

**Comment by pastaq** [comment]
> On a bad state I get happy triggers in evtest:
> 
> ```
> Event: time 1725338725.656849, -------------- SYN_REPORT ------------
> Event: time 1725338728.468155, type 1 (EV_KEY), code 708 (BTN_TRIGGER_HAPPY5), value 1
> Event: time 1725338728.468155, -------------- SYN_REPORT ------------
> Event: time 1725338728.654430, type 1 (EV_KEY), code 708 (BTN_TRIGGER_HAPPY5), value 0
> Event: time 1725338728.654430, -------------- SYN_REPORT ------------
> Event: time 1725338764.021574, type 1 (EV_KEY), code 710 (BTN_TRIGGER_HAPPY7), value 1
> Event: time 1725338764.021574, -------------- SYN_REPORT ------------
> Event: time 1725338764.207614, type 1 (EV_KEY), code 710 (BTN_TRIGGER_HAPPY7), value 0
> ```

These trigger events are from the default InputPlumber profile. The OpenGamepadUI profile isn't getting set every time. 

We're working on switching to a rust backend for our dbus interface (from cpp) which will enable us to respond to signaling instead of polling. It seems in rare cases we are still missing some events with the current system.

**Comment by philmmanjaro** [comment]
Reverting this change [75f312a](https://github.com/ShadowBlip/InputPlumber/commit/75f312a3b45c06e49c906f4441ae5ab78c41e224) from InputPlumber [0.34.5](https://github.com/ShadowBlip/InputPlumber/releases/tag/v0.34.5) makes no difference in the result. SteamUI acts similar, on evtest the other cmds are triggered. Looking forward to that rust backend. Do we have an rough ETA on that topic?

**Comment by pastaq** [comment]
> Reverting this change [75f312a](https://github.com/ShadowBlip/InputPlumber/commit/75f312a3b45c06e49c906f4441ae5ab78c41e224) from InputPlumber [0.34.5](https://github.com/ShadowBlip/InputPlumber/releases/tag/v0.34.5) makes no difference in the result. SteamUI acts similar, on evtest the other cmds are triggered. Looking forward to that rust backend. Do we have an rough ETA on that topic?

It's a big change to a lot of subsystems so it won't be rushed. Maybe a few weeks.

**Comment by philmmanjaro** [comment]
Could we add `BTN_TRIGGER_HAPPY7` and `BTN_TRIGGER_HAPPY5` to the LC/RC functions on the NEO in a temp workaround by checking if the other functions are not given and trigger a profile reload?

**Comment by pastaq** [comment]
> Could we add `BTN_TRIGGER_HAPPY7` and `BTN_TRIGGER_HAPPY5` to the LC/RC functions on the NEO in a temp workaround by checking if the other functions are not given and trigger a profile reload?

That won't solve the problem and is missing the point. Those events occur because of the profile that is set. There is more that happens besides the profile switching, that is just a symptom of the problem of InputPlumber and OpenGamepadUI not communicating properly. We also set the intercept events and intercept mode at the same time, which won't be working either.

**Comment by philmmanjaro** [comment]
I found out on how to trigger it since I had a typo in my `os-session-select` script:

```
  gamescope|steam|gamescope-session-steam)
    session_launcher="gamescope-session-steam.desktop"
    create_sentinel=1
  ;;
  steam-plus|gamescope-session-gamescope|steam-plus)
    session_launcher="gamescope-session-steam-plus.desktop"
    create_sentinel=1
  ;;
```

So a cold  boot would use `gamescope` which is linked just to the regular `gamescope-session-steam` which wouldn't load the **plus** bits. After fixing that and getting the controller out of the bad state by spawning a full OGUI session and going back to the steam-plus session it worked more stable. Have to monitor it. But this is a way to trigger it on cold boot.
